### DISCUSSION TITLE
repository_dispatch response unpredictable

### DISCUSSION THREAD

**Discussion by geekodour** [community/community#25045] [Actions]
<p>I am facing an interesting problem with <code>repository_dispatch</code>,</p>
<p>using it for a repo inside an GitHub org., when I use “my” access token, it’s <code>204 No Content</code>, i.e a success,</p>
<p>but when I use token of any other member(same membership level as mine) in the repo, it’s a 404 Not Found.</p>
<p>The scopes are all the same for the generated token.</p>
<p>Any idea what could be happening?</p>
<pre><code>curl -i -X POST -H "Authorization: token &lt;TOKEN&gt;" -H "Accept: application/vnd.github.everest-preview+json" -H "Content-Type: application/json" --data '{"event_type":"add_user","client_payload":{"key":"value"}}' "https://api.github.com/repos/&lt;ORG_NAME&gt;/&lt;REPO_NAME&gt;/dispatches"
</code></pre>
<p>Basically, just changing the <code>TOKEN</code> in the above snippet changes the behavior, even though the membership level and scopes are the same for another member in the org.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/geekodour">@geekodour</a> ,</p>
<p>I can repro on my side. This is due to organization team member is lack of ‘write’ permisson to the repository.</p>
<p>It’s ‘Read’ only by default, which means can not use ‘POST’ in rest api command. Please change it to ‘write’,  it works on my side.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/54e0a706c99d784ee985c291f598929521e7217e.png" data-download-href="/uploads/short-url/c6RqeEnqjHETPM2Ba49Q3D7ENf8.png?dl=1" title="Memberpermisson.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181088546-4bfb6955-4c50-4c1d-a356-e9cb14cc07c2.png" alt="Memberpermisson.png" data-base62-sha1="c6RqeEnqjHETPM2Ba49Q3D7ENf8" width="679" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/54e0a706c99d784ee985c291f598929521e7217e_2_679x500.png, https://aws1.discourse-cdn.com/github/original/2X/5/54e0a706c99d784ee985c291f598929521e7217e.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/5/54e0a706c99d784ee985c291f598929521e7217e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/54e0a706c99d784ee985c291f598929521e7217e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Memberpermisson.png</span><span class="informations">1004×739 46.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
